File: kotlin-ide/j2k/new/testData/newJ2k/assignmentExpression/inExpressionAssignmentsChain.java
class Test {
    public static void main(String[] args) {
        boolean a = true;
        boolean b = true;
        boolean c = true;
        boolean d = true;
        boolean e = true;

        if (a = b |= c = d &= e) ;
        while (a = b) ;
        do {} while (a ^= b);
        System.out.println(a &= b = c);
        System.out.println(a = b != c);
    }
}

File: kotlin-ide/j2k/new/testData/newJ2k/assignmentExpression/inExpressionAssignmentsChain.kt
internal object Test {
    @JvmStatic
    fun main(args: Array<String>) {
        var a = true
        var b = true
        var c = true
        var d = true
        val e = true
        if (e.let { d = d and it; d }.also { c = it }.let { b = b or it; b }.also { a = it });
        while (b.also { a = it });
        do {
        } while (b.let { a = a xor it; a })
        println(c.also { b = it }.let { a = a and it; a })
        println(b != c.also { a = it })
    }
}